src/main/scala/karazin/scala/users/group/week2/homework/Homework.scala

```diff
@@ -34,19 +34,24 @@ object Homework:
       !(this < that)
 
     @targetName("addition")
-    infix def +(that: Rational): Rational = ???
+    infix def +(that: Rational): Rational =
+      Rational(this.numer*that.denom+that.numer*this.denom,this.denom*that.denom)
 
     @targetName("negation")
-    infix def unary_- : Rational = ???
+    infix def unary_- : Rational =
+      -this
 
     @targetName("substraction")
-    infix def -(that: Rational): Rational = ???
+    infix def -(that: Rational): Rational =
+      Rational(this.numer*that.denom-that.numer*this.denom,this.denom*that.denom)
 
     @targetName("multiplication")
-    infix def *(that: Rational): Rational = ???
+    infix def *(that: Rational): Rational =
+      Rational(this.numer*that.numer,this.denom*that.denom)
 
     @targetName("devision")
-    infix def /(that: Rational): Rational = ???
+    infix def /(that: Rational): Rational =
```

Comment:
I'm pretty sure you haven't run the test.
Your code fails for a negative `that.numer`.